controllers/watcherdecisionengine_controller.go

```diff
@@ -43,17 +67,255 @@ type WatcherDecisionEngineReconciler struct {
+		return ctrl.Result{}, nil
+	}
+
+	// Handle service delete
+	if !instance.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, instance, helper)
+	}
+
+	configVars := make(map[string]env.Setter)
+	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
+	Log.Info(fmt.Sprintf("[DecisionEngine] Get secret 1 '%s'", instance.Spec.Secret))
+	secretHash, result, secret, err := ensureSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.Secret},
+		[]string{
+			instance.Spec.PasswordSelectors.Service,
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		r.RequeueTimeout,
+	)
+	if (err != nil || result != ctrl.Result{}) {
+		return result, err
+	}
+
+	configVars[instance.Spec.Secret] = env.SetValue(secretHash)
+
+	// all our input checks out so report InputReady
+	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
+
+	memcached, err := ensureMemcached(ctx, helper, instance.Namespace, instance.Spec.MemcachedInstance, &instance.Status.Conditions)
+
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Add finalizer to Memcached to prevent it from being deleted now that we're using it
+	if controllerutil.AddFinalizer(memcached, helper.GetFinalizer()) {
+		err := helper.GetClient().Update(ctx, memcached)
+		if err != nil {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				condition.MemcachedReadyCondition,
+				condition.ErrorReason,
+				condition.SeverityWarning,
+				condition.MemcachedReadyErrorMessage,
+				err.Error()))
+			return ctrl.Result{}, err
+		}
+	}
+
+	err = r.generateServiceConfigs(ctx, instance, secret, memcached, helper, &configVars)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)
+
+	// We reached the end of the Reconcile, update the Ready condition based on
+	// the sub conditions
+	if instance.Status.Conditions.AllSubConditionIsTrue() {
+		instance.Status.Conditions.MarkTrue(
+			condition.ReadyCondition, condition.ReadyMessage)
+	}
 
 	return ctrl.Result{}, nil
+
+}
+
+func (r *WatcherDecisionEngineReconciler) initStatus(instance *watcherv1beta1.WatcherDecisionEngine) error {
+
+	cl := condition.CreateList(
+		// Mark ReadyCondition as Unknown from the beginning, because the
+		// Reconcile function is in progress. If this condition is not marked
+		// as True and is still in the "Unknown" state, we `Mirror(` the actual
+		// failure/in-progress operation
+		condition.UnknownCondition(condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage),
+		condition.UnknownCondition(condition.InputReadyCondition, condition.InitReason, condition.InputReadyInitMessage),
+		condition.UnknownCondition(condition.ServiceConfigReadyCondition, condition.InitReason, condition.ServiceConfigReadyMessage),
```

Comment:
s/ServiceConfigReadyMessage/ServiceConfigInitMessage (autocomplete made the same mistake for me :) )

Reply:
ack but ServiceConfigReadyInitMessage... doing